cockroach/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,127 @@
+
+import sqlalchemy.types as sqltypes
+
+from .ddl_compiler import DDLCompiler
+from .type_compiler import TypeCompiler
+
+# Map type names (as returned by SHOW COLUMNS) to sqlalchemy type
+# objects.
+#
+# TODO(bdarnell): test more of these. The stock test suite only covers
+# a few basic ones.
+_type_map = dict(
+    boolean=sqltypes.BOOLEAN,
+    int=sqltypes.INT,
+    integer=sqltypes.INT,
+    smallint=sqltypes.INT,
+    bigint=sqltypes.INT,
+    float=sqltypes.FLOAT,
+    real=sqltypes.FLOAT,
+    double=sqltypes.FLOAT,
+    decimal=sqltypes.DECIMAL,
+    numeric=sqltypes.DECIMAL,
+    date=sqltypes.DATE,
+    timestamp=sqltypes.TIMESTAMP,
+    timestamptz=sqltypes.TIMESTAMP,
+    interval=sqltypes.Interval,
+    string=sqltypes.VARCHAR,
+    char=sqltypes.VARCHAR,
+    varchar=sqltypes.VARCHAR,
+    bytes=sqltypes.BLOB,
+    blob=sqltypes.BLOB,
+)
+
+
+class CockroachDialect(PGDialect_psycopg2):
+    name = 'cockroach'
+    supports_sequences = False
+
+    def __init__(self, *args, **kwargs):
+        super(CockroachDialect, self).__init__(*args,
+                                               use_native_hstore=False,
+                                               server_side_cursors=False,
+                                               **kwargs)
+
+    def initialize(self, connection):
+        # Bypass PGDialect's initialize implementation, which looks at
+        # server_version_info and performs postgres-specific queries
+        # to detect certain features on the server. Set the attributes
+        # by hand and hope things don't change out from under us too
+        # often.
+        super(PGDialect, self).initialize(connection)
+        self.implicit_returning = True
+        self.supports_native_enum = False
+        self.supports_smallserial = False
+        self._backslash_escapes = False
+
+    def _get_server_version_info(self, conn):
+        # PGDialect expects a postgres server version number here,
+        # although we've overridden most of the places where it's
+        # used.
+        return (9, 5, 0)
+
+    def _get_default_schema_name(self, conn):
+        raise NotImplementedError()
+
+    ddl_compiler = DDLCompiler
+    type_compiler = TypeCompiler
+
+    def get_table_names(self, conn, schema=None, **kw):
+        return [row.Table for row in conn.execute("SHOW TABLES")]
+
+    def has_table(self, conn, table, schema=None):
+        for i in self.get_table_names(conn, schema=schema):
+            if i == table:
+                return True
+        return False
+
+    def get_columns(self, conn, table_name, schema=None, **kw):
+        res = []
+        # TODO(bdarnell): escape table name
+        for row in conn.execute("SHOW COLUMNS FROM \"%s\"" % table_name):
+            name, type_name, nullable, default = row
+            # TODO(bdarnell): when there are type parameters, attach
+            # them to the returned type object.
+            typ = _type_map[type_name.split('(')[0].lower()]
```

Comment:
It contains every type we support today. For future-proofing, I've added an exception handler to fall back to `NULLTYPE`, which is what the postgres dialect does.